File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/LinkLabel/Functionality/ClientModePage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.LinkLabel.Functionality
{
    public partial class ClientModePage : UserControl
    {
        public ClientModePage()
        {
            InitializeComponent();

            // Update Label text
            this.mobjStatusLabel.Text = string.Format("Current client-mode status: {0}", this.mobjLinkLabel.ClientMode);
        }

        /// <summary>
        /// Handles the CheckedChanged event of the mobjCheckBox control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        void mobjCheckBox_CheckedChanged(object sender, System.EventArgs e)
        {
            // Set ClientMode property value for LinkLabel
            this.mobjLinkLabel.ClientMode = this.mobjCheckBox.Checked;
            // Update Label text
            this.mobjStatusLabel.Text = string.Format("Current client-mode status: {0}", this.mobjLinkLabel.ClientMode);
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/LinkLabel/Functionality/ClientModePage.vb
Namespace CompanionKit.Controls.LinkLabel.Functionality

	Public Class ClientModePage

        Public Sub New()
            ' This call is required by the Visual WebGui UserControl Designer.
            InitializeComponent()
            ' Update Label text
            Me.mobjStatusLabel.Text = String.Format("Current client-mode status: {0}", Me.mobjLinkLabel.ClientMode)
        End Sub

        ''' <summary>
        ''' Handles the CheckedChanged event of the mobjCheckBox control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjCheckBox_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles mobjCheckBox.CheckedChanged
            ' Set ClientMode property value for LinkLabel
            Me.mobjLinkLabel.ClientMode = Me.mobjCheckBox.Checked
            ' Update Label text
            Me.mobjStatusLabel.Text = String.Format("Current client-mode status: {0}", Me.mobjLinkLabel.ClientMode)
        End Sub

	End Class

End Namespace